packages/flutter/test/material/popup_menu_test.dart

```diff
@@ -1278,6 +1278,219 @@ void main() {
+    );
+    expect(
+      tester.getRect(find.widgetWithText(menuItemType, 'Item 2')).center.dy,
+      tester.getRect(find.text('Item 2')).center.dy,
+    );
+  });
+
+
+  testWidgets('CheckedPopupMenuItem custom padding', (WidgetTester tester) async {
+    final Key popupMenuButtonKey = UniqueKey();
+    final Type menuItemType = const CheckedPopupMenuItem<String>(child: Text('item')).runtimeType;
+
+    await tester.pumpWidget(
+      MaterialApp(
+        home: Scaffold(
+          body: Center(
+            child: PopupMenuButton<String>(
+              key: popupMenuButtonKey,
+              child: const Text('button'),
+              onSelected: (String result) { },
+              itemBuilder: (BuildContext context) {
+                return <PopupMenuEntry<String>>[
+                  const CheckedPopupMenuItem<String>(
+                    padding: EdgeInsets.zero,
+                    value: '0',
+                    child: Text('Item 0'),
+                  ),
+                  const CheckedPopupMenuItem<String>(
+                    padding: EdgeInsets.zero,
+                    height: 0,
+                    value: '0',
+                    child: Text('Item 1'),
+                  ),
+                  const CheckedPopupMenuItem<String>(
+                    padding: EdgeInsets.all(20),
+                    value: '0',
+                    child: Text('Item 2'),
+                  ),
+                  const CheckedPopupMenuItem<String>(
+                    padding: EdgeInsets.all(20),
+                    height: 100,
+                    value: '0',
+                    child: Text('Item 3'),
+                  ),
+                ];
+              },
+            ),
+          ),
+        ),
+      ),
+    );
+
+    // Show the menu
+    await tester.tap(find.byKey(popupMenuButtonKey));
+    await tester.pumpAndSettle();
+
+    // The menu items and their InkWells should have the expected vertical size
+    // given the interactions between heights and padding.
+    expect(tester.getSize(find.widgetWithText(menuItemType, 'Item 0')).height, 56); // Minimum ListTile height (56)
+    expect(tester.getSize(find.widgetWithText(menuItemType, 'Item 1')).height, 56); // Minimum ListTile height (56)
+    expect(tester.getSize(find.widgetWithText(menuItemType, 'Item 2')).height, 96); // Padding (20.0 + 20.0) + Height of ListTile (56) = 96
+    expect(tester.getSize(find.widgetWithText(menuItemType, 'Item 3')).height, 100); // Height value of 100, since child (56) + padding (40) < 100
+
+    expect(tester.widget<Container>(find.widgetWithText(Container, 'Item 0')).padding, EdgeInsets.zero);
+    expect(tester.widget<Container>(find.widgetWithText(Container, 'Item 1')).padding, EdgeInsets.zero);
+    expect(tester.widget<Container>(find.widgetWithText(Container, 'Item 2')).padding, const EdgeInsets.all(20));
+    expect(tester.widget<Container>(find.widgetWithText(Container, 'Item 3')).padding, const EdgeInsets.all(20));
+  });
+
+
```

Comment:
nit, an extra new line

Reply:
oops, sorry i missed this before i merged. i made a follow-up PR for it, along with other simple changes: https://github.com/flutter/flutter/pull/78755